app.ts

```diff
@@ -22,11 +22,15 @@ class AjaxCall {
 	}
 
 	onRStateChange(callback: Function) {
-		if (this.objHttpReq.readyState === 4 && this.objHttpReq.status === 200) {
-			let data = JSON.parse(this.objHttpReq.responseText);
-			(callback)(data, this.indexStart, this.indexFinish);
+		if (this.objHttpReq.readyState === XMLHttpRequest.DONE) {
+			if (this.objHttpReq.status === 200) {
+				let data = JSON.parse(this.objHttpReq.responseText);
+				(callback)(data, this.indexStart, this.indexFinish);
+			}
+			else {
+				console.log("AJAX error with the following status occurred: " + this.objHttpReq.status);
```

Comment:
We never you console.log. It just causes problems and populate console with useless information. We dont even use it for debugging because one tend to forget to remove it when done.